RepoInspector/RepoInspector.Runner/Program.cs

```diff
@@ -138,139 +116,72 @@ static async Task Main(string[] args)
-            
+#endif
+
             return GetProdJsonSerializerSettings(scorerFactory);
         }
 
         private static JsonSerializerSettings GetDebugJsonSerializerSettings(IScorerFactory scorerFactory)
         {
-            return new JsonSerializerSettings
+            return new()
             {
                 ContractResolver = new CamelCasePropertyNamesContractResolver(),
                 DefaultValueHandling = DefaultValueHandling.Include,
                 NullValueHandling = NullValueHandling.Include,
                 Formatting = Formatting.Indented,
                 DateFormatHandling = DateFormatHandling.IsoDateFormat,
-                Converters = new List<JsonConverter> { new StringEnumConverter(), new TruncatingDoubleConverter(), new ScorerConverter(scorerFactory) },
+                Converters = new List<JsonConverter> { new StringEnumConverter(), new TruncatingDoubleConverter(), new ScorerConverter(scorerFactory) }
             };
         }
 
         private static JsonSerializerSettings GetProdJsonSerializerSettings(IScorerFactory scorerFactory)
         {
-            return new JsonSerializerSettings
+            return new()
             {
                 ContractResolver = new CamelCasePropertyNamesContractResolver(),
                 Formatting = Formatting.None,
                 DefaultValueHandling = DefaultValueHandling.Ignore,
                 NullValueHandling = NullValueHandling.Ignore,
                 DateFormatHandling = DateFormatHandling.IsoDateFormat,
-                Converters = new List<JsonConverter> { new StringEnumConverter(), new TruncatingDoubleConverter(), new ScorerConverter(scorerFactory) },
+                Converters = new List<JsonConverter> { new StringEnumConverter(), new TruncatingDoubleConverter(), new ScorerConverter(scorerFactory) }
             };
         }
 
-        private static void Console_CancelKeyPress(object sender, ConsoleCancelEventArgs e)
-        {
-            e.Cancel = true;
-            HandleShutdown();
-        }
-
-        private static void HandleShutdown(AssemblyLoadContext ctx)
-            => HandleShutdown();
-
-        private static void HandleShutdown()
-        {
-            _logger.LogInformation("SIGTERM received. Shutting down");
-            
-            var stopwatch = Stopwatch.StartNew();
-            _cts.Cancel();
-            stopwatch.Stop();
-
-            _logger.LogInformation($"Daemon stopped in {stopwatch.ElapsedMilliseconds:N0}ms");
-        }
-        
-        private static IScorerFactory GetKnownScorers(IApprovalAnalyzer approvalAnalyzer, IWordCounter wc)
+        private static IScorerFactory GetKnownScorers(IServiceProvider sp)
         {
             var scorers = GetDerivedTypes<Scorer>(Assembly.GetAssembly(typeof(Scorer)));
             var scorerInstances = scorers
-                .Select(s => CreateInstance(s, approvalAnalyzer, wc))
+                .Select(s => ActivatorUtilities.GetServiceOrCreateInstance(sp, s))
```

Comment:
Am I crazy? Where is this implementation?

Reply:
Yeah, that's why this is WIP. `ActivatorUtilities` is provided by the framework, and I *think* there's a way to make it do the magic, but the whole instantiation of scorers is a little unclear to me -- it seems like given the current setup, they'll always get GitHub-flavored parsers?

Reply:
* Scorers are repo-type agnostic.
* PR readers are repo-type-specific

Each PR reader is instantiated by the `PullRequestReaderFactory`, looking at the type of `RepositoryKind` in each `WatchedRepo`.